src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptCompletionServiceWithProviders.cs

```diff
@@ -25,20 +24,23 @@ internal sealed override CompletionRules GetRules(CompletionOptions options)
 
         internal sealed override void FilterItems(
            Document document,
-           IReadOnlyList<(CompletionItem, PatternMatch?)> itemsWithPatternMatch,
+           IReadOnlyList<MatchResult> matchResults,
            string filterText,
-           IList<CompletionItem> builder)
-            => FilterItemsImpl(document, itemsWithPatternMatch, filterText, builder);
+           IList<MatchResult> builder)
+            => FilterItemsImpl(document, matchResults, filterText, builder);
 
         internal virtual void FilterItemsImpl(
```

Comment:
TS isn't using this at the moment, so we can change it w/o breaking them. We should move them off the public one though since it's not performant

Reply:
I have verified that TS completion is still working properly with this PR

Reply:
FYI @MariaSolOs, something I think TS should switch to once this is inserted for better perf.